specification/eventgrid/data-plane/Microsoft.DataBox/stable/2018-01-01/DataBox.json

```diff
@@ -0,0 +1,74 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-01-01",
+    "title": "Schema of Azure Data Box events published to Azure Event Grid",
+    "description": "Describes the schema of the Azure Data Box events published to Azure Event Grid. This corresponds to the Data property of an EventGridEvent."
+  },
+  "paths": {},
+  "definitions": {
+    "DataBoxCopyStartedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.DataBox.CopyStarted event.",
+      "type": "object",
+      "properties": {
+        "serialNumber": {
+          "description": "Serial Number of the device associated with the event, separated by comma delimiter in case of multiple",
+          "type": "string"
+        },
+        "stageDetails": {
+          "$ref": "#/definitions/StageDetail"
+        }
+      }
+    },
+    "DataBoxCopyCompletedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.DataBox.CopyCompleted event.",
+      "type": "object",
+      "properties": {
+        "serialNumber": {
+          "description": "Serial Number of the device associated with the event, separated by comma delimiter in case of multiple",
+          "type": "string"
+        },
+        "stageDetails": {
+          "$ref": "#/definitions/StageDetail"
+        }
+      }
+    },
+    "DataBoxOrderCompletedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.DataBox.OrderCompleted event.",
+      "type": "object",
+      "properties": {
+        "serialNumber": {
+          "description": "Serial Number of the device associated with the event, separated by comma delimiter in case of multiple",
+          "type": "string"
+        },
+        "stageDetails": {
+          "$ref": "#/definitions/StageDetail"
+        }
+      }
+    },
+    "StageDetail": {
```

Comment:
I would suggest removing this type and just inlining the properties in each model. Eventually, we will have support for modularizing common properties without needing to copy/paste.

Reply:
I have updated it, can you please check

Reply:
The type is still being defined - you are just defining it multiple times. I'm suggesting that we completely remove "stageDetails" from the swagger and define the properties directly in each model.

Reply:
Hi, like discussed over the teams, we will be keeping the complex object of stage details, because we might expose more data in future not related to stage details.

Reply:
Talked offline - @aakash049 expects that there will be new non-stage specific properties added in the future so it is more clear to keep the stage properties separate.